src/nvim/buffer_defs.h

```diff
@@ -649,7 +649,7 @@ struct file_buffer {
   bool b_p_swf;                 /* 'swapfile' */
   long b_p_smc;                 /* 'synmaxcol' */
   char_u      *b_p_syn;         /* 'syntax' */
-  long b_p_ts;                  /* 'tabstop' */
+  int b_p_ts;                   // 'tabstop'
```

Comment:
do we already have any numeric option that is not `long` ? If not then I don't think we can be sure that it is safe, for instance in `option.c` all numeric options are read/written as `*(long*)`.